concordium-consensus/src/Concordium/GlobalState/Persistent/Trie.hs

```diff
@@ -76,16 +83,86 @@ instance OrdFixedTrieKey Word32
 deriving via Word64 instance OrdFixedTrieKey BakerId
 deriving via Word64 instance OrdFixedTrieKey DelegatorId
 
+-- |A pair of an index and a branch.
+data BranchEntry r = BranchEntry
+    { -- |The branch index.
+      beIndex :: {-# UNPACK #-} !Word8,
+      -- |The branch itself.
+      beBranch :: !r
+    }
+    deriving (Show, Functor, Foldable, Traversable)
+
+-- |Data structure representing the branches of the Trie.
+newtype Branches r = Branches
+    { -- |The branches, represented as an ordered array of pairs of indices and non-null values.
+      theBranches :: Array.Array (BranchEntry r)
+    }
+    deriving (Show, Functor, Foldable, Traversable)
+
+-- |Convert 'Branches' to a list. The list will have length 256.
```

Comment:
I'd put an explanation here that 256 is the alphabet set cardinality, and not some random number.